### DISCUSSION TITLE
Daybreak (or other Xerox hardware) support?

### DISCUSSION THREAD

**Discussion by eschaton** [Interlisp/medley#1862] [Implementation history]
Does current Medley support the Xerox Daybreak (1186 or 6085) hardware, or any of the other old Xerox hardware? Obviously most users will still be using the virtual machine, but it'd be cool if the updated system could still run on old hardware.

**Comment by nbriggs** [comment]
Back when I was working on Lisp things at PARC, IIRC the guidance was that you could bring a sysout forward from a D-machine to a system running with the (Maiko) emulator, but you could not take a sysout that had been run on the Maiko emulator back to a D-machine.  When the image is started it detects that it's on the Maiko emulator and smashes a bunch of functions to replace them with the hooks into the emulator compiled code [e.g., for access to the Unix filesystem rather than the D-machine's local file system code]- and it's not set up to do that undoably.  My opinion:  It would be an interesting project, for someone who wanted to dig in to the low-level guts of the system - not for the faint of heart -- and takes a certain sensibility and an appreciation for the way things were done, and no outspoken opinions on how they "ought to be fixed to do it right" rather than making the repairs to restore the existing functionality.
